hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/Statistic.java

```diff
@@ -207,6 +209,12 @@
       "S3Guard metadata store put one metadata path latency"),
   S3GUARD_METADATASTORE_INITIALIZATION("s3guard_metadatastore_initialization",
       "S3Guard metadata store initialization times"),
+  S3GUARD_METADATASTORE_RECORD_READS(
+      "s3guard_metadatastore_record_reads",
+      "S3Guard metadata store records read"),
+  S3GUARD_METADATASTORE_RECORD_WRITES(
+      "s3guard_metadatastore_record_writes",
+      "S3Guard metadata store records written"),
```

Comment:
Seems like this could be interesting for comparing different rename trackers? If so I'm curious what sort of stats / deltas you get with the new  logic.

Reply:
Good point. With the ancestor tracking the new renamer shouldn't do any more than before. And I can verify that the commit operations don't do duplicate puts of all the parents, the way they do currently